Remove call to rangy.init()

iainbeeston opened the pull request:
On firefox, if I have a page that loads textAngular, but textAngular is
not used at all (or if I load a textAngular element dynamically after
page load), rangy.init() fails. This prevents rangy from ever
initializing, which cripples textAngular.

Here's the stacktrace (shortened slightly):

    "Init listener threw an exception. Continuing." TypeError: sel is null
    Stack trace:
    @http://localhost:5000/assets/rangy/rangy-core.js?body=1:2533:13
    @http://localhost:5000/assets/rangy/rangy-core.js?body=1:2513:10
    window.rangy</api.createModule/<@http://localhost:5000/assets/rangy/rangy-core.js?body=1:236:13
    init@http://localhost:5000/assets/rangy/rangy-core.js?body=1:167:17
    @http://localhost:5000/assets/textAngular/textAngular.js?body=1:1833:3

Interestingly, rangy.init() is only called to test that
rangy.saveSelection is loaded. This is unnecessary - rangy initializes
saveSelection when it loads, and removing the call to rangy.init() does
not affect the check for saveSelection (so far as I can tell).
Furthermore, textAngular 1.2.x never called rangy.init(), but instead relied on
rangy to initialize itself.

To fix the bug I've removed the rangy.init() call. This means we can't
easily check if the SaveSelection module is loaded, but I think it's
fair to assume that if some has read enough of the readme to install
rangy, they'll also have read enough to install SaveSelection.

iainbeeston commented:
This happens on textAngular 1.3.11, rangy 1.2.3 and Firefox 37

PhilFlash commented:
+1
Same problem on Firefox 36.0.4 and Internet Explorer 11.0.9600.17690 
(with textAngular 1.3.11 and rangy 1.2.3)
The problem is solved by this patch on Firefox and Internet Explorer.
Thanks

iainbeeston commented:
I didn't even realise this happened on IE11 as well...
On Mon, 30 Mar 2015 at 5:58 pm, PhilFlash <notifications@github.com> wrote:

> +1
> Same problem on Firefox 36.0.4 and Internet Explorer 11.0.9600.17690
> (with textAngular 1.3.11 and rangy 1.2.3)
> The problem is solved by this patch on Firefox and Internet Explorer.
> Thanks
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/fraywing/textAngular/pull/601#issuecomment-87750350>.
>

SimeonC commented:
Hey, thanks for the PR and sorry for the delay - Been really busy.

One question, is there any reason you removed the `if(!$window.rangy.saveSelection){` check? TextAngular does break down a bit if you don't have the saveSelection module not loaded.

iainbeeston commented:
After I removed `$window.rangy.init()`, `$window.rangy.saveSelection` is never defined at that point (unless textAngular is already rendered onto the page). I also tried checking `rangy.modules`, but selection save and restore apparently wasn't there either.

So without a better way to check, I just removed that code. Hopefully people will add both rangy and save selection and restore at the same time (and not one but not the other)

iainbeeston commented:
Is there any chance of getting this resolved? At present I'm stuck on angular 1.2 and textAngular 1.2 because of this issue (textAngular 1.2.2 doesn't have the issue, but it also doesn't allow angular 1.3 in it's bower.json)

jaisonerick commented:
I had the same issue at #638 and just submitted a PR at #639. The problem was with rangy itself but was fixed in the 1.3.x version of it.

iainbeeston commented:
Upgrading to rangy 1.3 worked for me, thanks for the tip.

However, rangy 1.3 is still in beta. I have no problem using it in production, but others may.

SimeonC commented:
Thinking about this - and looking through the rangy repo - I think I'm going to take #639 option and upgrade. It looks like they're nearly at release as their roadmap for 1.3 is complete and from memory they said the core and saveselection were stable, it just was the newer modules that were beta state.

As a side note this change should have been done in the lib folder not the src folder.

Closing in favor of updating rangy to 1.3.0
